packages/dialtone-vue2/common/storybook_utils.js

```diff
@@ -5,18 +5,21 @@ import iconNames from '@dialpad/dialtone-icons/dist/icons.json';
  * This is useful for more complex components that are hard to work with and
  * get messy when rendered via a template string. Will pass args and argTypes
  * into the component so it is able to be live edited with storybook controls addon.
- * @param {object} args storybook control args
- * @param {object} argTypes storybook control argument settings
- * @param {component} templateComponent vue component template for rendering to storybook.
- *                                      Note this should not be the component itself,
- *                                      but rather the usage of that component.
+ * @param component this will get the component name
+ * @param defaultTemplate we will mount in this component
+ * @param argsData storybook control args
  * @returns {component} the template component with props and args added.
  */
-export const createTemplateFromVueFile = (args, argTypes, templateComponent) => ({
-  components: { templateComponent },
-  props: Object.keys(argTypes),
-  template: '<template-component v-bind="$props"></template-component>',
-});
+
+export function createRenderConfig (component, defaultTemplate, argsData) {
+  return {
+    components: { [component.name]: defaultTemplate },
+    setup () {
+      return { argsData };
+    },
+    template: `<${component.name} v-bind="argsData"></${component.name}>`,
+  };
+}
```

Comment:
If we're just using the name of the component, it's not easier and clearer to pass the name instead of the component?

Reply:
As how we are using the function e.g:
`render: (argsData) => createRenderConfig(DtBanner, BannerDefault, argsData),`

I see it better than manually pass the name of the component, kind of unstable and sensitive to typo.

Reply:
makes sense, thanks for the clarification